### DISCUSSION TITLE
Append automatically selects appended text

### DISCUSSION THREAD

**Discussion by davxy** [helix-editor/helix#8851] [General]
Is expected for the text inserted after that the append key (`a`) is pressed to be automatically selected?

If yes, I personally find this a bit surprising (as inserted text is also highlighted). Is there a way to disable this?

For example the append to the end of line (`A`) doesn't automatically selects the inserted text.

**Comment by davxy** [comment]
After some experimentation the question now is: makes sense to maintain the selection when entering insert mode?
If yes, what are the use cases?

**Comment by David-Else** [comment]
I just noticed this, I don't remember it being like that before. If it doesn't select on insert then surely it should not on append? Also the inconsistency with `a` and `A` seems really bad.

**Comment by the-mikedavis** [comment]
This has been discussed previously, see https://github.com/helix-editor/helix/issues/3744#issuecomment-1240726094 or https://github.com/helix-editor/helix/discussions/2924
